### PR TITLE
feat(stok lot)csv download considers filters

### LINKED ISSUE #2638: Downloading CSV of stock registry should limit to only the searched items
When you download a CSV from the stock registry, the entire registry is printed out.  If an enterprise has been functioning a long time, it will have a ton of stock and we should allow them to only download part of the list.  Additionally, most other registries function this way.

Steps to reproduce:
 1. Navigate to the Stock Registry.
 2. Search on a depot.
 3. Click "Download as CSV".
 4. Note that the entire list is downloaded.

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #2638

### CODE REVIEW

**Comment by jeremielodi** [comment]
@jniles, what test I should write for this issue, it should concern the csv file content or grid

**Comment by jniles** [comment]
@jeremielodi, good question!

The easiest thing: You can write an integration test that pull data using the `json` renderer.  The code you have changed affects all renderers, not just the CSV.  So you can do two tests:
 1. The first will look at `/reports/stock/lots?renderer=json` and make sure it downloads the full set of rows.
 2. The second test will look at `/reports/stock/lots?depot_uuid=XXXX?renderer=json` and make sure it downloads only part of the dataset.

Since all renderers begin with the JSON renderer, we can assume that, if the JSON renderer limits the date, the rest of them will also limit the data.

File: test/integration/stock.js [hunk 1]
```diff
@@ -183,6 +183,28 @@ describe('(/stock/) The Stock HTTP API', () => {
...
+    () => agent.get(`/reports/stock/lots?renderer=json`)
+      .then((res) => {
+        expect(res.body.rows.length).to.equal(12);
+      })
+      .catch(helpers.handler)
+  );
+  
+  // (report) render all stock exit relatives to 'Depot Principal'
+  it(
+    `GET /reports/stock/lots?renderer=json
```
Comment (jniles) [inline test/integration/stock.js:199]: This is the right idea!

The text should actually reflect what the test is testing.  So, in this case, it should read:
```js
it(`GET /reports/stock/lots?renderer=json&depot_uuid=${depotPrincipalUuid} returns exists for depot principal (10 out)`)
```

**Review by jniles — APPROVED** [review]
Really good!  One last update to the text so that we all know what the test is doing and we can merge this.

**Pull request merged**